### PR TITLE
Feature/df 979 remove candidate email

### PR DESCRIPTION
**Pull request by fabiangotzen**

### CODE REVIEW

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java [hunk 1]
```diff
@@ -99,6 +97,14 @@ public CandidateDto getOne(UUID id) throws CandidateNotFoundException {
         return this.candidateMapper.toCandidateDto(this.getCandidate(id));
     }
 
+    @Transactional(readOnly = true)
+    @PreAuthorize("@authorizationService.canViewProtectedData(#id)")
+    public CandidateAnonymousContactInfo getCandidateAnonymousContactInfo(UUID id) throws AnonymousContactInfoNotFoundException {
+        log.debug("Request to get Candidate Mail for: {}", id);
+        Optional<Candidate> candidateToContact = this.candidateRepository.findById(id);
+        return candidateToContact.map(candidate -> new CandidateAnonymousContactInfo().setEmail(candidate.getEmail()).setExternalId(candidate.getExternalId())).orElseThrow(() -> new AnonymousContactInfoNotFoundException(id));
```
Comment (birom) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java]: Please fix formatting. Single line is hardly readable. Thx.

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java [hunk 2]
```diff
@@ -99,6 +97,14 @@ public CandidateDto getOne(UUID id) throws CandidateNotFoundException {
         return this.candidateMapper.toCandidateDto(this.getCandidate(id));
     }
 
+    @Transactional(readOnly = true)
+    @PreAuthorize("@authorizationService.canViewProtectedData(#id)")
```
Comment (birom) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java]: Should be only available for``ROLE_ADMIN``.

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateAnonymousContactInfo.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package ch.admin.seco.service.candidate.service.dto;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotNull;
+
+public class CandidateAnonymousContactInfo {
+
+    @NotNull
```
Comment (birom) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateAnonymousContactInfo.java]: There is no need for validation this DTO is only for reading the data.

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java (hunk 1 above)
Comment (fabiangotzen) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java]: fixed

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java (hunk 2 above)
Comment (fabiangotzen) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/CandidateService.java]: used @isAdmin annotation instead

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateAnonymousContactInfo.java (hunk 3 above)
Comment (fabiangotzen) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateAnonymousContactInfo.java]: removed

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/AnonymousContactInfoNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package ch.admin.seco.service.candidate.service;
+
+import java.util.UUID;
+
+public class AnonymousContactInfoNotFoundException extends Exception {
```
Comment (birom) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/AnonymousContactInfoNotFoundException.java]: I think this should be a ``RuntimeException`` and we also need to add it to the ``ExceptionTranslator``.

Comment (fabiangotzen) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/AnonymousContactInfoNotFoundException.java]: as discussed, used CandidateNotFoundException instead

**Pull request merged**